test/integration/short/control-connection-simulator-tests.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const assert = require('assert');
+const simulacron = require('../simulacron');
+const utils = require('../../../lib/utils');
+const types = require('../../../lib/types');
+
+const Client = require('../../../lib/client.js');
+
+describe('ControlConnection', function() {
+  this.timeout(5000);
+
+  before(simulacron.start.bind(simulacron));
+  after(simulacron.stop.bind(simulacron));
+
+  describe.skip("#init", function() {
+    it('should downgrade to protocol v3 with versions 3.0 & 2.1', testWithNodes(['3.0.13', '2.1.17'], 3));
+    it('should downgrade to protocol v3 with versions 2.2 & 2.1', testWithNodes(['2.2.11', '2.1.17'], 3));
+    it('should downgrade to protocol v2 with versions 2.2 & 2.0', testWithNodes(['2.2.11', '2.0.17'], 2));
+    it('should downgrade to protocol v1 with versions 2.2 & 1.2', testWithNodes(['2.2.11', '1.2.19'], 1));
+    it('should downgrade to protocol v2 with versions 2.1 & 2.0', testWithNodes(['2.1.17', '2.0.17'], 2));
+    it('should downgrade to protocol v1 with versions 2.1 & 1.2', testWithNodes(['2.1.17', '1.2.19'], 1));
+    it('should downgrade to protocol v1 with versions 2.0 & 1.2', testWithNodes(['2.0.17', '1.2.19'], 1));
+    // no need to downgrade since both support protocol V4.
+    it('should not downgrade with versions 3.0 & 2.2', testWithNodes(['3.0.13', '3.0.11', '2.2.9'], 4));
+    // can't downgrade because C* 3.0 does not support protocol V2.
+    it('should not downgrade with versions 3.0 & 2.0', testWithNodes(['3.0.13', '2.0.17'], 4));
```

Comment:
The only difference between "Can't downgrade" test and "No need to downgrade" test, from output perspective, is the log warning. Should verify it?

Reply:
Good idea, I can check for that.